src/FastRouteRouter.php

```diff
@@ -411,7 +438,7 @@ private function marshalMatchedRoute(array $result, string $method): RouteResult
         $params = $result[2];
 
         $options = $route->getOptions();
-        if (! empty($options['defaults'])) {
+        if (isset($options['defaults']) && is_array($options['defaults'])) {
```

Comment:
Might need to check `!== []` here as empty would have checked for that

Reply:
An empty array here would have no consequences